NDTensors/ext/NDTensorsCUDAExt/linearalgebra.jl

```diff
@@ -11,3 +11,7 @@ function NDTensors.svd_catch_error(A::CuMatrix; alg="JacobiAlgorithm")
   end
   return USV
 end
+
+function NDTensors.mul!!(::Type{<:CuArray}, CM, AM, BM, α, β)
```

Comment:
I think to be more forward-thinking this should be:
```julia
function NDTensors.mul!!(::Type{<:CuArray}, CM, ::Type{<:CuArray}, AM, ::Type{<:CuArray}, BM, α, β)
  # ...
end
```
i.e. dispatch on the `leaf_parenttype` of each of the inputs. Then we can catch more specific situations like if one input is on GPU and one on CPU and do specialized behavior on those cases (for example throw errors or move to CPU or GPU depending on the memory available on the GPU).